src/main.ts

```diff
@@ -82,7 +87,8 @@ async function run(inputs: Inputs): Promise<void> {
     const deploy = await netlifyClient.deploy(siteId, deployFolder, {
       draft: isDraft,
       message: deployMessage,
-      configPath: netlifyConfigPath
+      configPath: netlifyConfigPath,
+      branch: getBranchName() // https://github.com/netlify/js-client/releases/tag/v4.3.0
```

Comment:
could this also take in the branch from inputs?
`const branch: string | undefined = inputs.branch()`
```suggestion
      branch: branch || getBranchName() // https://github.com/netlify/js-client/releases/tag/v4.3.0
```